Automattic/jetpack
# Decouple Publicize Connection Lists from UI Take 2

Take 2 of the excellent #9955 by @c-shultz. See #9039.

Updated to work on top of #10043.

#### Changes proposed in this Pull Request:
* Adds a method to `Publicize_Base` to get the list of Publicize Connections suitable for use in UI.
* Cleans up logic v. presentation boundary in `Publicize_UI`.

#### Testing instructions:
##### Automated Tests
1. `yarn docker:phpunit --filter=WP_Test_Publicize`

##### Manual Tests
1. Under an *admin* account, set up a test site with two publicize accounts: one "global" (available to all users of the site), one not global.
2. As that *admin*, go to the Classic (non-Gutenberg) editor, and look at the Publicize UI for a new post.
3. You should see two checked checkboxes: one for each connection. You should be able to uncheck each (the inputs should not be disabled).
4. Publish the post.
5. You should see the publicized content (tweet, FB update, etc.) on both connections.
6. Under an *author* account, go to the Classic (non-Gutenberg) editor, and look at the Publicize UI or a new post.
7. You should see one checked-but disabled checkbox for the global connection. You should not see any evidence at all of the admin's non-global connection.
8. Publish a post as that *author*.
9. See the publicized content (tweet, FB update, etc.) on the global connection.

#### Proposed changelog entry for your changes:
None? Part of a larger initiative to provide Publicize data via REST API.
